### PR TITLE
[SERVICE-834, SERVICE-819] Package and sign asar

### PR DESCRIPTION
**Pull request by wenjunche**

### CODE REVIEW

File: src/scripts/createProviderZip.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export function createProviderZip() {
     const archive = archiver('zip', {zlib: {level: 9}});
 
     output.on('close', () => {
-        console.log(`Zip file created at '${output.path}'`);
+        console.log(`Zip file created at xxx '${output.path}'`);
```
Comment (tim-dinsdale) [inline src/scripts/createProviderZip.ts]: Looks left over?

Comment (pjbroadbent) [inline src/scripts/createProviderZip.ts]: Revert this?

File: src/scripts/createAsar.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    ].forEach((file) => {
+        fs.copyFileSync(file, path.resolve(output, path.basename(file)));
+    });
+
+    [
+        ...glob.sync(path.resolve(output, '**', '*'), {dot: true})
+    ].forEach((file) => {
+        if (!fs.statSync(file).isDirectory() && !file.endsWith('.ofds')) {
+            console.log(`signing ${file}`);
+            openfinSign(file);
```
Comment (pjbroadbent) [inline src/scripts/createAsar.ts:38]: The sample code in the ticket looks like it builds the asar and then only signs the asar.

This is signing every file in the asar, then separately signing the asar itself. Is that intentional?..

File: src/scripts/createAsar.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        path.resolve(getRootDirectory(), 'res', 'provider')
+    ].forEach((file) => {
+        fs.copySync(file, output);
+    });
+
+    [
+        path.resolve(getRootDirectory(), 'dist', 'client', `openfin-${SERVICE_NAME}.js`)
+    ].forEach((file) => {
+        fs.copyFileSync(file, path.resolve(output, path.basename(file)));
+    });
```
Comment (pjbroadbent) [inline src/scripts/createAsar.ts:32]: Does the `asar` lib support streams? Would ideally create the archive without local file copying, as per `createProviderZip.ts`.

**Review by tim-dinsdale — CHANGES_REQUESTED** [review]

File: src/scripts/createAsar.ts (hunk 3 above)
Comment (wenjunche) [reply src/scripts/createAsar.ts:32]: asar does not support streams.  Also,  each file needs to be signed, in addition to fdc3.asar.

File: src/scripts/createAsar.ts (hunk 2 above)
Comment (wenjunche) [reply src/scripts/createAsar.ts:38]: Yes

**Review by pjbroadbent — APPROVED** [review]

**Review by tim-dinsdale — APPROVED** [review]

**Pull request merged**